Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -120,36 +116,32 @@ void CEXIETHERNET::BuiltInBBAInterface::HandleDHCP(const Common::UDPPacket& pack
   to.sin_family = IPPROTO_UDP;
   to.sin_port = udpdata.source_port;
 
-  Common::DHCPBody reply;
-
   const std::vector<u8> ip_part = {((u8*)&m_router_ip)[0], ((u8*)&m_router_ip)[1],
                                    ((u8*)&m_router_ip)[2], ((u8*)&m_router_ip)[3]};
+  const std::vector<u8> time = {0, 1, 0x51, 0x80};
 
-  response.ip_header = Common::IPv4Header(308, IPPROTO_UDP, from, to);
-
-  response.udp_header = Common::UDPHeader(from, to, 300);
-
-  reply = Common::DHCPBody(request.transaction_id,
-                           *(Common::MACAddress*)&m_eth_ref->mBbaMem[BBA_NAFR_PAR0], m_current_ip,
-                           m_router_ip);
+  Common::DHCPView reply;
+  reply.body =
+      Common::DHCPBody(request.transaction_id,
+                       Common::BitCastPtr<Common::MACAddress>(&m_eth_ref->mBbaMem[BBA_NAFR_PAR0]),
+                       m_current_ip, m_router_ip);
 
   // options
-  request.options[2] == 1 ? reply.AddDHCPOption(1, 53, std::vector<u8>{2}) :
-                            reply.AddDHCPOption(1, 53, std::vector<u8>{5});
-  reply.AddDHCPOption(4, 54, ip_part);                            // dhcp server ip
-  reply.AddDHCPOption(4, 51, std::vector<u8>{0, 1, 0x51, 0x80});  // lease time 24h
-  reply.AddDHCPOption(4, 58, std::vector<u8>{0, 1, 0x51, 0x80});  // renewal
-  reply.AddDHCPOption(4, 59, std::vector<u8>{0, 1, 0x51, 0x80});  // rebind
-  reply.AddDHCPOption(4, 1, std::vector<u8>{255, 255, 255, 0});   // submask
-  reply.AddDHCPOption(4, 28,
-                      std::vector<u8>{ip_part[0], ip_part[1], ip_part[2], 255});  // broadcast ip
-  reply.AddDHCPOption(4, 6, ip_part);                                             // dns server
-  reply.AddDHCPOption(3, 15, std::vector<u8>{0x6c, 0x61, 0x6e});  // domaine name "lan"
-  reply.AddDHCPOption(4, 3, ip_part);                             // router ip
-  reply.AddDHCPOption(0, 255, {});                                // end
-
-  response.data =
-      std::vector<u8>((u8*)&reply, (u8*)&reply + 300);  // 300 is the size with current options
+  view.options[0].at(2) == 1 ? reply.AddOption(1, 53, {2}) : reply.AddOption(1, 53, {5});
```

Comment:
You should probably check the size of `options` and `options[0]` to make sure they are valid _(since you switched from array to vector)_.